### DISCUSSION TITLE
Does go agent support service group

### DISCUSSION THREAD

**Discussion by xiaozaiyuji** [apache/skywalking#11164] [Q&A] [answered]
![image](https://github.com/apache/skywalking/assets/4230320/6753ede5-1c0e-4559-bec3-a31cfe88ef40)

The first env is not config.
As the picture,I config the go agent like this,but the service is not show in the ui.
Does this is not support service group? If the config is wrong,what should I do?
![image](https://github.com/apache/skywalking/assets/4230320/48c6b695-cf0e-4a39-b018-c00445ebbcb0)

![image](https://github.com/apache/skywalking/assets/4230320/313c68f9-c2e2-4032-8a1a-fae1f75f595b)

**Comment by wu-sheng** [comment]
The service name is the only concept in the agent side. There is never a group concept. The group name is just a service name prefix.
So, whatever you want to set the service name, relies on your local setup.

Don't create a thing by imagination please.

**Reply by xiaozaiyuji** [reply]
In agent I can echo this env param.I think this is not wrong.I did not konw where is wrong.

**Reply by xiaozaiyuji** [reply]
![image](https://github.com/apache/skywalking/assets/4230320/46a6aaf8-68f9-4253-9d38-289cd2fb4a5a)

I config the env SW_AGENT_SERVICE_NAME console::cmdb::dev。 It shown .I think this may be something wrong.

**Reply by wu-sheng** [reply]
There is never a doc saying we support your way to link several system env variables together. That is what I mean you just image.
One system env for your service name. Done. Nothing else.

**Reply by xiaozaiyuji** [reply]
![image](https://github.com/apache/skywalking/assets/4230320/fa106597-95fb-4827-be34-cc012a92290b)
In java this support.I config this for go,I will do more test.

**Reply by wu-sheng** [reply, marked as answer]
I don't know Java supports this. If so, it is just by accident.  Anyway, this is never official.

**Reply by xiaozaiyuji** [reply]
![image](https://github.com/apache/skywalking/assets/4230320/bc15e3bc-0e28-460d-9e48-fdb222726048)

This will effect and show the servicename in my k8s env,But I think this is something we should do to improve.The go agent is effective.I will use this in product environment. I will gether more information for you.

**Reply by xiaozaiyuji** [reply]
![image](https://github.com/apache/skywalking/assets/4230320/b23f645a-d566-4938-b27d-cd70e30729c6)
After run for a while.I was so confusion for this the service instance is not shown in console::cmdb::dev,but show in 
![image](https://github.com/apache/skywalking/assets/4230320/545a8380-6de1-4e4c-a208-263287df6a17)
I did not know why this happen.This is some cache?

**Reply by xiaozaiyuji** [reply]
I rollback this config to service_name: ${SW_AGENT_NAME:}. In order to support env get.
